### ISSUE TITLE
insert csv file can not set settings

### ISSUE THREAD

**Issue by ZzMark** [ClickHouse/clickhouse-java#1246]
using clickhouse-jdbc-client 0.4.0
```java
String insertTableName = "import_test";
File csvFile = new File("input.csv.gz");
final ClickHouseNode endpoint = ClickHouseNode.of("http://127.0.0.1:8123/default");
try (
        ClickHouseClient client = ClickHouseClient.builder()
                .nodeSelector(ClickHouseNodeSelector.of(ClickHouseProtocol.HTTP))
                .build();
        ClickHouseResponse response = client.connect(endpoint)
                .write()
                .table(insertTableName)
                .data(ClickHouseFile.of(csvFile))
                .set("input_format_csv_skip_first_lines", 1) // skip first line
                .format(ClickHouseFormat.CSV)
                .executeAndWait()
) {
    System.out.println("success");
} catch (ClickHouseException e) {
    throw new RuntimeException(e);
}
```

err:
```
com.clickhouse.client.ClickHouseException: Expect one SQL statement to execute but we got 2, server ClickHouseNode [uri=http://127.0.0.1:8123/default]@1346537010
	at com.clickhouse.client.ClickHouseException.of(ClickHouseException.java:168)
	at com.clickhouse.client.ClickHouseClient.executeAndWait(ClickHouseClient.java:896)
	at com.clickhouse.client.ClickHouseRequest.executeAndWait(ClickHouseRequest.java:2068)
	at com.volcana.CHUtil.insertFromCsvFile(CHUtil.java:105)
Caused by: java.lang.IllegalArgumentException: Expect one SQL statement to execute but we got 2
	at com.clickhouse.client.http.ClickHouseHttpClient.send(ClickHouseHttpClient.java:98)
	at com.clickhouse.client.AbstractClient.sendAsync(AbstractClient.java:158)
	at com.clickhouse.client.AbstractClient.lambda$execute$0(AbstractClient.java:291)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run$$$capture(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by zhicwu** [comment]
Sorry for the late response. The client needs to be enhanced to support your case. Meanwhile, perhaps you can try `request.option(ClickHouseClientOption.CUSTOM_SETTINGS, "input_format_csv_skip_first_lines=1")` or `request.option(ClickHouseHttpOption.CUSTOM_HTTP_PARAMS, "input_format_csv_skip_first_lines=1")` instead of `set("input_format_csv_skip_first_lines", 1)`?

**Comment by ZzMark** [comment]
is same.

ClickhouseHttpClient.send:
```sql
SET input_format_csv_skip_first_lines='1'
INSERT INTO bi_3.test_left_1
 FORMAT CSV
```

when CUSTOM_SETTINGS
`option()` same set()

https://github.com/ClickHouse/clickhouse-java/blob/4a6940b7de33c87e6ddc65de1c1aad258f3270fb/clickhouse-client/src/main/java/com/clickhouse/client/ClickHouseRequest.java#L990-L1022

**Comment by Dong-GuoSheng** [comment]
> is same.
> 
> ClickhouseHttpClient.send:
> 
> ```sql
> SET input_format_csv_skip_first_lines='1'
> INSERT INTO bi_3.test_left_1
>  FORMAT CSV
> ```
> 
> when CUSTOM_SETTINGS `option()` same set()
> 
> https://github.com/ClickHouse/clickhouse-java/blob/4a6940b7de33c87e6ddc65de1c1aad258f3270fb/clickhouse-client/src/main/java/com/clickhouse/client/ClickHouseRequest.java#L1010-L1017

你好，这个问题解决了吗？

我也遇到类似的问题了，我想导入tsv文件，想忽略错误行，使用了client和jdbc设置都没用。
jdbc设置：
 conn.setClientInfo("CustomHttpParameters", "input_format_allow_errors_num=100");
 conn.setClientInfo("custom_http_params", "input_format_allow_errors_num=100");
try (Statement stmt = conn.createStatement()) {
                    ClickHouseRequest<?> request = stmt.unwrap(ClickHouseRequest.class);
                    request.set("custom_http_params", "input_format_allow_errors_num=100");
                    request.set("input_format_allow_errors_num", 100);
                    request.set("input_format_allow_errors_ratio", 0.5);
                    request.option(ClickHouseClientOption.CUSTOM_SETTINGS, "input_format_allow_errors_num=100");
设置没有生效，
参考ClickHouseClient.send通过多个sql set变量也没有生效。

**Comment by mzitnik** [comment]
@zhicwu closing this issue already was solved #1258
